backend/src/main/java/com/empire/Geography.java

```diff
@@ -0,0 +1,79 @@
+package com.empire;
+
+import com.google.gson.GsonBuilder;
+import java.io.FileReader;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class Geography {
+	private static final transient Map<Integer, Map<Integer, Geography>> cache = new ConcurrentHashMap<>();
+
+	List<Region> regions;
+	List<Border> borders;
+	List<Kingdom> kingdoms;
+	int players;
+	int holycity;
+
+	static class Region {
+		int id;
+		com.empire.Region.Type type;
+		String name;
+		int core;
+	}
+
+	static class Border {
+		int a;
+		Integer b; // b can be omitted for map boundary borders.
+		int w;
+		List<Point> path;
+	}
+
+	static class Point {
+		double x;
+		double y;
+	}
+
+	static class Kingdom {
+		String name;
+		Culture culture;
+	}
+
+	static Geography loadGeography(int ruleSetId, int numPlayers) throws IOException {
+		Map<Integer, Geography> ruleGeography = cache.computeIfAbsent(ruleSetId, unused -> new ConcurrentHashMap<>());
+		if (!ruleGeography.containsKey(numPlayers)) {
+			ruleGeography.put(numPlayers, readGeography(ruleSetId, numPlayers));
+		}
+		return ruleGeography.get(numPlayers);
+	}
+
+	private static Geography readGeography(int ruleSetId, int numPlayers) throws IOException {
+		try (FileReader r = new FileReader("resources/rules/" + ruleSetId + "/maps/" + numPlayers + ".json")) {
+			return new GsonBuilder().create().fromJson(r, Geography.class);
+		} catch (FileNotFoundException e) {
+			try (InputStreamReader r = new InputStreamReader(Geography.class.getResourceAsStream("/rules/" + ruleSetId + "/maps/" + numPlayers + ".json"))) {
```

Comment:
Yeah, that is one piece of the style guide I dislike - Google uses 80 character for C++ (which is terrible), 100 for Java (better, still tight), and no limit for Go (best, IMO). So I will totally admit to picking and choosing from my favorites here. If you'd prefer we start using a 120c limit we could do that, but then we should probably replace all \t with two spaces or something.

(The other major intentional departure from the style guide is that I prefer to in-line single-statement blocks, e.g.:
if (condition) statement;
to 
if (condition) {
  statement;
}
which is sort of because of my experiences using Ruby.)